# Coleta de Dados Estruturados

A base de dados está disponível no seguinte link: https://www.kaggle.com/gpreda/covid-world-vaccination-progress

In [1]:
#Importando as bibliotecas
import pandas as pd
import os

In [2]:
diretorio = "C:\\IGTI_Data_Science\\Vacinados\\dataset\\"
arquivo = "country_vaccinations.csv"
dados = os.path.join(diretorio,arquivo)

In [3]:
dados

'C:\\IGTI_Data_Science\\Vacinados\\dataset\\country_vaccinations.csv'

In [4]:
dataframe = pd.read_csv(dados,sep=',',encoding = 'latin-1')

In [5]:
dataframe.head()

country iso_code        date  total_vaccinations  people_vaccinated  \
0  Albania      ALB  2021-01-10                 0.0                0.0   
1  Albania      ALB  2021-01-11                 NaN                NaN   
2  Albania      ALB  2021-01-12               128.0              128.0   
3  Albania      ALB  2021-01-13               188.0              188.0   
4  Albania      ALB  2021-01-14               266.0              266.0   

   people_fully_vaccinated  daily_vaccinations_raw  daily_vaccinations  \
0                      NaN                     NaN                 NaN   
1                      NaN                     NaN                64.0   
2                      NaN                     NaN                64.0   
3                      NaN                    60.0                63.0   
4                      NaN                    78.0                66.0   

   total_vaccinations_per_hundred  people_vaccinated_per_hundred  \
0                            0.00                           0.00   
1                             NaN                            NaN   
2                            0.00                           0.00   
3                            0.01                           0.01   
4                            0.01                           0.01   

   people_fully_vaccinated_per_hundred  daily_vaccinations_per_million  \
0                                  NaN                             NaN   
1                                  NaN                            22.0   
2                                  NaN                            22.0   
3                                  NaN                            22.0   
4                                  NaN                            23.0   

          vaccines         source_name  \
0  Pfizer/BioNTech  Ministry of Health   
1  Pfizer/BioNTech  Ministry of Health   
2  Pfizer/BioNTech  Ministry of Health   
3  Pfizer/BioNTech  Ministry of Health   
4  Pfizer/BioNTech  Ministry of Health   

                                      source_website  
0  https://shendetesia.gov.al/covid19-ministria-e...  
1  https://shendetesia.gov.al/covid19-ministria-e...  
2  https://shendetesia.gov.al/covid19-ministria-e...  
3  https://shendetesia.gov.al/covid19-ministria-e...  
4  https://shendetesia.gov.al/covid19-ministria-e...

# Analisando os dados


## Sobre o dataset

O dataset utilizado é referente a estatistica do número de vacinados da COVID-19 em todos os países do mundo.

Para esse estudo vamos utilizar as váriaveis: 
*country = pais;
date = data;
total_vaccinations = Número total de vacinações (número absoluto de imunizações totais no país);
daily_vaccinations = Número de vacinados por dia;
Vaccines = vacinas utilizadas no país;
;*


Vamos responder as perguntas:

1. Quais são os países com maior número de pessoas imunizadas?
2. Qual vacina é a mais utilizada no mundo? ** Existem diferentes tipos de vacinas disponível no mundo
3. Qual a país possui o maior número de vacinados por dia?
4. Qual a média de vacinação de vacinação por pais?

Após responder as perguntas vamos exportar as respostas obtidas.

In [13]:
#Verificando dados referente ao brasil por dia

#dataframe.loc[dataframe['country']=="Brazil"]

dataframe.query('country=="Brazil"').sort_values('date',ascending=False)[:5]

country iso_code        date  total_vaccinations  people_vaccinated  \
573  Brazil      BRA  2021-02-26           8101787.0          6346769.0   
572  Brazil      BRA  2021-02-25           7799000.0          6202055.0   
571  Brazil      BRA  2021-02-24           7551676.0          6116082.0   
570  Brazil      BRA  2021-02-23           7297061.0          6002873.0   
569  Brazil      BRA  2021-02-22           7028356.0          5857080.0   

     people_fully_vaccinated  daily_vaccinations_raw  daily_vaccinations  \
573                1755018.0                302787.0            223804.0   
572                1596945.0                247324.0            227474.0   
571                1435594.0                254615.0            238305.0   
570                1294188.0                268705.0            241018.0   
569                1171276.0                 77554.0            247768.0   

     total_vaccinations_per_hundred  people_vaccinated_per_hundred  \
573                            3.81                           2.99   
572                            3.67                           2.92   
571                            3.55                           2.88   
570                            3.43                           2.82   
569                            3.31                           2.76   

     people_fully_vaccinated_per_hundred  daily_vaccinations_per_million  \
573                                 0.83                          1053.0   
572                                 0.75                          1070.0   
571                                 0.68                          1121.0   
570                                 0.61                          1134.0   
569                                 0.55                          1166.0   

                        vaccines                                  source_name  \
573  Oxford/AstraZeneca, Sinovac  Regional governments via Coronavirus Brasil   
572  Oxford/AstraZeneca, Sinovac  Regional governments via Coronavirus Brasil   
571  Oxford/AstraZeneca, Sinovac  Regional governments via Coronavirus Brasil   
570  Oxford/AstraZeneca, Sinovac  Regional governments via Coronavirus Brasil   
569  Oxford/AstraZeneca, Sinovac  Regional governments via Coronavirus Brasil   

                         source_website  
573  https://coronavirusbra1.github.io/  
572  https://coronavirusbra1.github.io/  
571  https://coronavirusbra1.github.io/  
570  https://coronavirusbra1.github.io/  
569  https://coronavirusbra1.github.io/

In [15]:
#Verificando o número de países distintos na base
dataframe['country'].nunique()

112

Temos 112 registros de Países no dataset

# 1.Quais são os países com maior número de pessoas imunizadas?

In [16]:
#Identificando a data mais recente da coleta
ultima_data = dataframe['date'].max()
ultima_data

'2021-02-26'

In [18]:
#Filtrando a última data
ultima_data = dataframe[dataframe.date == ultima_data]

In [24]:
ultima_data.shape

(43, 15)

Com essa informação podemos verificar que nem todos os países estão contabilizados na contagem da última data, pois temos apenas 43 linhas no dataframe e verificamos que temos 112 países diferentes registrados.
Para resolver o problema, precisamos pegar o último registro computado de cada país.

In [25]:
dataframe.groupby('country')['date'].max().sort_values(ascending=False)

country
Zimbabwe           2021-02-26
Hungary            2021-02-26
Lebanon            2021-02-26
Romania            2021-02-26
Portugal           2021-02-26
                      ...    
Myanmar            2021-02-02
Egypt              2021-01-30
Greenland          2021-01-27
Northern Cyprus    2021-01-22
Monaco             2021-01-18
Name: date, Length: 112, dtype: object

In [28]:
dataframe.groupby('country')['total_vaccinations'].max().sort_values(ascending=False)

country
United States          70454064.0
China                  40520000.0
United Kingdom         19913592.0
England                16785841.0
India                  13756940.0
                          ...    
Paraguay                   1000.0
New Zealand                1000.0
Trinidad and Tobago         440.0
Venezuela                   157.0
Saint Helena                107.0
Name: total_vaccinations, Length: 112, dtype: float64

In [29]:
imunizados_por_pais = dataframe.groupby('country')['total_vaccinations'].max().sort_values(ascending=False)
imunizados_por_pais

country
United States          70454064.0
China                  40520000.0
United Kingdom         19913592.0
England                16785841.0
India                  13756940.0
                          ...    
Paraguay                   1000.0
New Zealand                1000.0
Trinidad and Tobago         440.0
Venezuela                   157.0
Saint Helena                107.0
Name: total_vaccinations, Length: 112, dtype: float64

In [30]:
type(imunizados_por_pais)

pandas.core.series.Series

Para transformar novamente em um dataframe temos que usar o comando to_frame() e resetar o index.

In [31]:
imunizados = imunizados_por_pais.to_frame().reset_index()
imunizados

country  total_vaccinations
0          United States          70454064.0
1                  China          40520000.0
2         United Kingdom          19913592.0
3                England          16785841.0
4                  India          13756940.0
..                   ...                 ...
107             Paraguay              1000.0
108          New Zealand              1000.0
109  Trinidad and Tobago               440.0
110            Venezuela               157.0
111         Saint Helena               107.0

[112 rows x 2 columns]

In [33]:
imunizados.sort_values('total_vaccinations',ascending=False).head(5)

country  total_vaccinations
0   United States          70454064.0
1           China          40520000.0
2  United Kingdom          19913592.0
3         England          16785841.0
4           India          13756940.0

# 2.Qual vacina é a mais utilizada no mundo?

In [39]:
vacinas_por_pais = dataframe[['country','date','vaccines']]
vacinas_por_pais

country        date           vaccines
0      Albania  2021-01-10    Pfizer/BioNTech
1      Albania  2021-01-11    Pfizer/BioNTech
2      Albania  2021-01-12    Pfizer/BioNTech
3      Albania  2021-01-13    Pfizer/BioNTech
4      Albania  2021-01-14    Pfizer/BioNTech
...        ...         ...                ...
4375  Zimbabwe  2021-02-22  Sinopharm/Beijing
4376  Zimbabwe  2021-02-23  Sinopharm/Beijing
4377  Zimbabwe  2021-02-24  Sinopharm/Beijing
4378  Zimbabwe  2021-02-25  Sinopharm/Beijing
4379  Zimbabwe  2021-02-26  Sinopharm/Beijing

[4380 rows x 3 columns]

In [35]:
vacinas_por_pais.groupby('vaccines')['country'].nunique().sort_values(ascending=False)

vaccines
Pfizer/BioNTech                                                                       26
Moderna, Oxford/AstraZeneca, Pfizer/BioNTech                                          20
Oxford/AstraZeneca                                                                    15
Sputnik V                                                                             11
Oxford/AstraZeneca, Pfizer/BioNTech                                                   10
Moderna, Pfizer/BioNTech                                                               7
Sinopharm/Beijing                                                                      6
Sinovac                                                                                2
Oxford/AstraZeneca, Sinopharm/Beijing                                                  2
Pfizer/BioNTech, Sinovac                                                               2
Pfizer/BioNTech, Sinopharm/Beijing, Sputnik V                                          2
Sinopharm/Be

In [40]:
vacinas_pais = dataframe.groupby('country')['vaccines'].unique()
vacinas_pais = vacinas_pais.to_frame().reset_index()

In [41]:
vacinas_pais

country                               vaccines
0           Albania                      [Pfizer/BioNTech]
1           Algeria                            [Sputnik V]
2           Andorra                      [Pfizer/BioNTech]
3          Anguilla                   [Oxford/AstraZeneca]
4         Argentina                            [Sputnik V]
..              ...                                    ...
107  United Kingdom  [Oxford/AstraZeneca, Pfizer/BioNTech]
108   United States             [Moderna, Pfizer/BioNTech]
109       Venezuela                            [Sputnik V]
110           Wales  [Oxford/AstraZeneca, Pfizer/BioNTech]
111        Zimbabwe                    [Sinopharm/Beijing]

[112 rows x 2 columns]

26 países utilizando a vacina da Pfizer

# 3.Qual a país possui o maior número de vacinados por dia?

In [44]:
dataframe.groupby('country')['daily_vaccinations'].max().sort_values(ascending=False)

country
China                  1916190.0
United States          1716311.0
India                   472428.0
United Kingdom          445204.0
Turkey                  435596.0
                         ...    
Andorra                    100.0
Trinidad and Tobago         63.0
Venezuela                   31.0
Greenland                    NaN
Saint Helena                 NaN
Name: daily_vaccinations, Length: 112, dtype: float64

# 4.Qual a média de vacinação por pais?

In [45]:
media_pais = dataframe.groupby('country')['daily_vaccinations'].mean().sort_values(ascending=False)
media_pais = media_pais.to_frame().reset_index()
media_pais

country  daily_vaccinations
0     United States       961578.338235
1             China       623616.000000
2             India       298508.476190
3    United Kingdom       255473.256757
4           England       215682.081081
..              ...                 ...
107         Andorra           62.000000
108   Liechtenstein           52.619048
109       Venezuela           31.000000
110       Greenland                 NaN
111    Saint Helena                 NaN

[112 rows x 2 columns]

Gerando novos dados

In [47]:
#Renomeando algumas colunas
media_pais = media_pais.rename(columns={'country': 'pais', 'daily_vaccinations': 'media_vacinacao' })
media_pais

pais  media_vacinacao
0     United States    961578.338235
1             China    623616.000000
2             India    298508.476190
3    United Kingdom    255473.256757
4           England    215682.081081
..              ...              ...
107         Andorra        62.000000
108   Liechtenstein        52.619048
109       Venezuela        31.000000
110       Greenland              NaN
111    Saint Helena              NaN

[112 rows x 2 columns]

In [48]:
vacinas_pais = vacinas_pais.rename(columns={'country': 'pais', 'vaccines': 'vacina_utilizada'})
vacinas_pais[:5]

pais      vacina_utilizada
0    Albania     [Pfizer/BioNTech]
1    Algeria           [Sputnik V]
2    Andorra     [Pfizer/BioNTech]
3   Anguilla  [Oxford/AstraZeneca]
4  Argentina           [Sputnik V]

In [49]:
imunizados = imunizados.rename(columns={'country': 'pais', 'total_vaccinations': 'Imunizados'})
imunizados[:5]

pais  Imunizados
0   United States  70454064.0
1           China  40520000.0
2  United Kingdom  19913592.0
3         England  16785841.0
4           India  13756940.0

# Criando um novo dataset

In [50]:
dados_tratados = pd.merge(imunizados,media_pais,how='inner')

In [51]:
dados_tratados = pd.merge(dados_tratados,vacinas_pais)
dados_tratados

pais  Imunizados  media_vacinacao  \
0          United States  70454064.0    961578.338235   
1                  China  40520000.0    623616.000000   
2         United Kingdom  19913592.0    255473.256757   
3                England  16785841.0    215682.081081   
4                  India  13756940.0    298508.476190   
..                   ...         ...              ...   
107             Paraguay      1000.0       250.000000   
108          New Zealand      1000.0       108.000000   
109  Trinidad and Tobago       440.0        63.000000   
110            Venezuela       157.0        31.000000   
111         Saint Helena       107.0              NaN   

                                  vacina_utilizada  
0                       [Moderna, Pfizer/BioNTech]  
1    [Sinopharm/Beijing, Sinopharm/Wuhan, Sinovac]  
2            [Oxford/AstraZeneca, Pfizer/BioNTech]  
3            [Oxford/AstraZeneca, Pfizer/BioNTech]  
4                    [Covaxin, Oxford/AstraZeneca]  
..                                             ...  
107                                    [Sputnik V]  
108                              [Pfizer/BioNTech]  
109                           [Oxford/AstraZeneca]  
110                                    [Sputnik V]  
111                           [Oxford/AstraZeneca]  

[112 rows x 4 columns]

# Exportando dados tratados

In [54]:
dados_tratados.to_csv('estatistica_pais.csv', index=True, sep = ';')

In [53]:
dados_tratados.to_json('estatistica.json', orient = 'records')